core/src/main/scala/wop/game/WopState.scala

```diff
@@ -58,10 +62,20 @@ object WopState {
   sealed trait InProgress extends WopState {
     def player: Player
     def board: Board
+    def currentTime: Duration
+    def deadline: Duration
     def apply(point: Point): Either[Foul, WopState]
+    def withTimestamp(t: Duration): InProgress
+
+    def onTick: Either[Foul.Timeout, InProgress] = {
+      val time: Duration = now
+      if (time >= deadline) Left(Foul.Timeout(player))
+      else Right(withTimestamp(time))
+    }
   }
 
-  case class Turn(player: Player, board: Board, currentSubBoard: Point) extends InProgress {
+  case class Turn(player: Player, board: Board, currentSubBoard: Point, deadline: Duration, currentTime: Duration)
```

Comment:
I suppose `deadline` and `currentTime ` should be `Long`. `Duration` is not good to store timestamps.